src/Solocraft.cpp

```diff
@@ -77,106 +490,722 @@ class solocraft_player_instance_handler : public PlayerScript {
+			case 109: 
+				dunLevel = M109; break;
+			case 129: 
+				dunLevel = M129; break;
+			case 189: 
+				dunLevel = M189; break;
+			case 209: 
+				dunLevel = M209; break;
+			case 229: 
+				dunLevel = M229; break;
+			case 230: 
+				dunLevel = M230; break;
+			case 249: 
+				dunLevel = M249; break;
+			case 289: 
+				dunLevel = M289; break;
+			case 309: 
+				dunLevel = M309; break;
+			case 329: 
+				dunLevel = M329; break;
+			case 349: 
+				dunLevel = M349; break;
+			case 389: 
+				dunLevel = M389; break;
+			case 409: 
+				dunLevel = M409; break;
+			case 429: 
+				dunLevel = M429; break;
+			case 469: 
+				dunLevel = M469; break;
+			case 509: 
+				dunLevel = M509; break;
+			case 531: 
+				dunLevel = M531; break;
+			case 269: 
+				dunLevel = M269; break;
+			case 532: 
+				dunLevel = M532; break;
+			case 534: 
+				dunLevel = M534; break;
+			case 540: 
+				dunLevel = M540; break;
+			case 542: 
+				dunLevel = M542; break;
+			case 543: 
+				dunLevel = M543; break;
+			case 544: 
+				dunLevel = M544; break;
+			case 545: 
+				dunLevel = M545; break;
+			case 546: 
+				dunLevel = M546; break;
+			case 547: 
+				dunLevel = M547; break;
+			case 548: 
+				dunLevel = M548; break;
+			case 550: 
+				dunLevel = M550; break;
+			case 552: 
+				dunLevel = M552; break;
+			case 553: 
+				dunLevel = M553; break;
+			case 554: 
+				dunLevel = M554; break;
+			case 555: 
+				dunLevel = M555; break;		
+			case 556: 
+				dunLevel = M556; break;
+			case 557: 
+				dunLevel = M557; break;
+			case 558: 
+				dunLevel = M558; break;
+			case 560: 
+				dunLevel = M560; break;
+			case 564: 
+				dunLevel = M564; break;
+			case 565: 
+				dunLevel = M565; break;
+			case 568: 
+				dunLevel = M568; break;
+			case 580: 
+				dunLevel = M580; break;
+			case 585: 
+				dunLevel = M585; break;
+			case 533: 
+				dunLevel = M533; break;
+			case 574: 
+				dunLevel = M574; break;
+			case 575: 
+				dunLevel = M575; break;
+			case 576: 
+				dunLevel = M576; break;
+			case 578: 
+				dunLevel = M578; break;
+			case 595: 
+				dunLevel = M595; break;
+			case 599: 
+				dunLevel = M599; break;
+			case 600: 
+				dunLevel = M600; break;
+			case 601: 
+				dunLevel = M601; break;
+			case 602: 
+				dunLevel = M602; break;
+			case 603: 
+				dunLevel = M603; break;
+			case 604: 
+				dunLevel = M604; break;
+			case 608: 
+				dunLevel = M608; break;
+			case 615: 
+				dunLevel = M615; break;
+			case 616: 
+				dunLevel = M616; break;
+			case 619: 
+				dunLevel = M619; break;
+			case 631: 
+				dunLevel = M631; break;
+			case 632: 
+				dunLevel = M632; break;
+			case 649: 
+				dunLevel = M649; break;
+			case 650: 
+				dunLevel = M650; break;
+			case 658: 
+				dunLevel = M658; break;
+			case 668: 
+				dunLevel = M668; break;
+			case 724: 
+				dunLevel = M724; break;		
+			default:
+				dunLevel = SolocraftDungeonLevel;
+		}
+        return dunLevel; //return the specific dungeon's level 
+	}	
```

Comment:
if you create a hashmap you don't need all of this code

Reply:
exapmle:
```C++
#include <unordered_map>
std::unordered_map<uint32, uint32> dungeons = {
  { 33, sConfigMgr->GetIntDefault("Solocraft.ShadowfangKeep.Level", 15) }
};
```

and to use it as dunLevel you can just use
```C++
dunLevel = deungeons[map->GetId()];
```
or better, `return  deungeons[map->GetId()];`